src/pages/api/auth/[...nextauth].ts

```diff
@@ -1,15 +1,38 @@
+import createFetch from '@vercel/fetch'
 import NextAuth from 'next-auth'
 import CloudIdpProvider from 'lib/auth/cloud-idp-provider'
 import refreshTokenSet from 'lib/auth/refresh-token-set'
 import isJwtExpired from 'lib/auth/is-jwt-expired'
 import { AuthErrors } from 'types/auth'
 
+const fetch = createFetch()
+
 const isDev = process.env.NODE_ENV === 'development'
 
 export default NextAuth({
 	session: {
 		maxAge: 2592000, // 30 days
 	},
+	// https://next-auth.js.org/configuration/options#events
+	events: {
+		/**
+		 * NOTE: NextAuth does not log out of auth providers, so we have to handle doing that
+		 * ourselves in this signOut event.
+		 * https://github.com/nextauthjs/next-auth/discussions/3938
+		 */
+		async signOut({ token }) {
+			try {
+				await fetch(
+					`${__config.dev_dot.auth.idp_url}/oauth2/sessions/logout?id_token_hint=${token.id_token}`
```

Comment:
It would be best to get this url from the well known configuration (i.e. `${__config.dev_dot.auth.idp_url}/.well-known/openid-configuration`).

Reply:
That sounds like a great idea! I'll make that update